pkg/web/http.go

```diff
@@ -0,0 +1,69 @@
+package web
+
+import (
+	"bytes"
+	"crypto/tls"
+	"github.com/gorilla/websocket"
+	"io"
+	"log"
+	"net"
+	"net/http"
+	"time"
+)
+
+type WrappedSocketListener struct {
+	sockets chan *WrappedSocket
+	done    chan struct{}
+	addr    net.Addr
+}
+
+func (wsl *WrappedSocketListener) Accept() (net.Conn, error) {
+	select {
+	case ws := <-wsl.sockets:
+		return ws, nil
+	case <-wsl.done:
+		return nil, io.ErrClosedPipe
+	}
+}
+
+func (wsl *WrappedSocketListener) Close() error {
+	wsl.done <- struct{}{}
+	return nil
+}
+
+func (wsl *WrappedSocketListener) Addr() net.Addr {
+	return wsl.addr
+}
+
+var upgrader = websocket.Upgrader{
+	HandshakeTimeout: 20 * time.Second,
+	Subprotocols:     []string{"mumble1.2.5", "binary"},
```

Comment:
Should this be 1.2.4? IIRC, there were no protocol changes between 1.2.4 and 1.2.5.

Reply:
Preferably just "mumble". If we break compat, we'd just do "mumble2", "mumble3", etc. (or whatever)